joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -130,6 +132,41 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
     Seq(_callAst)
   }
 
+  protected def astForSimpleMemberAccessExpression(accessExpr: DotNetNodeInfo): Seq[Ast] = {
+    val fieldIdentifierName = nameFromNode(accessExpr)
+
+    val fieldInScope = scope.findFieldInScope(fieldIdentifierName)
+    val targetName =
+      if (fieldInScope.nonEmpty && fieldInScope.get.isStatic) scope.surroundingTypeDeclFullName.getOrElse(Defines.Any)
+      else "this"
+
+    val typeFullName   = fieldInScope.map(_.typeFullName).getOrElse(Defines.Any)
+    val identifierName = targetName
+    val identifier     = newIdentifierNode(identifierName, typeFullName)
+
+    val fieldIdentifier = NewFieldIdentifier()
+      .code(fieldIdentifierName)
+      .canonicalName(fieldIdentifierName)
+      .lineNumber(accessExpr.lineNumber)
+      .columnNumber(accessExpr.columnNumber)
+
+    val fieldAccessCode = s"$identifierName.$fieldIdentifierName"
```

Comment:
No, the `identifierName` gets changed based on whether the field is static or not. So for static access we have `<ClassName>.<fieldName>`, otherwise we have `this.<fieldName>`. `code(accessExpr)` always returns `this.<fieldName>`